Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -3677,65 +3689,55 @@ private ObservableList<ElementInfo> GetAllElementsFromPage(string path, List<eEl
-                        if (filteredElementType != null && filteredElementType.Count > 0)
+                        // set the Flag in case you wish to learn the element or not
+                        bool learnElement = true;
+
+                        //filter element if needed, in case we need to learn only the MappedElements .i.e., LearnMappedElementsOnly is checked
+                        if (filteredElementType != null)
                         {
+                            //Case Learn Only Mapped Element : set learnElement to false in case element doesn't exist in the filteredElementType List AND element is not frame element
                             if (!filteredElementType.Contains(elementTypeEnum.Item2))
+                                learnElement = false;
+                        }
+
+                        if (learnElement)
+                        {
+                            IWebElement el = Driver.FindElement(By.XPath(htmlElemNode.XPath));
+                            if (el == null)
                             {
                                 continue;
                             }
-                        }
 
-                        IWebElement el = Driver.FindElement(By.XPath(htmlElemNode.XPath));
-                        if (el == null)
-                        {
-                            continue;
-                        }
+                            //filter none visible elements
+                            if (!el.Displayed || el.Size.Width == 0 || el.Size.Height == 0)
+                            {
+                                continue;
+                            }
 
-                        //filter none visible elements
-                        if (!el.Displayed || el.Size.Width == 0 || el.Size.Height == 0)
-                        {
-                            continue;
-                        }
+                            HTMLElementInfo foundElemntInfo = GetElementInfo(path, htmlElemNode, elementTypeEnum, el);
 
-                        HTMLElementInfo foundElemntInfo;
+                            foundElemntInfo.IsAutoLearned = true;
+                            foundElementsList.Add(foundElemntInfo);
 
-                        if (learnPartialElementInfoDetails)
-                        {
-                            foundElemntInfo = new HTMLElementInfo();
-                            foundElemntInfo.ElementType = elementTypeEnum.Item1;
-                            foundElemntInfo.ElementTypeEnum = elementTypeEnum.Item2;
-                            foundElemntInfo.ElementObject = el;
-                            foundElemntInfo.Path = path;
-                            foundElemntInfo.XPath = htmlElemNode.XPath;
-                            foundElemntInfo.HTMLElementObject = htmlElemNode;
-                            ((IWindowExplorer)this).LearnElementInfoDetails(foundElemntInfo);
+                            allReadElem.Add(foundElemntInfo);
                         }
```

Comment:
validate changes work and nothing got broken

Reply:
validated ! working fine Meni